core/loadpoint.go

```diff
@@ -782,15 +791,32 @@ func (lp *Loadpoint) plannerActive() (active bool) {
 		return false
 	}
 
-	// if the plan did not (entirely) work, we may still be charging beyond plan end- in that case, continue charging
 	if active {
+		// ignore short plans if not already active
+		if !lp.planActive && requiredDuration < 10*time.Minute {
+			lp.log.DEBUG.Printf("plan too short- ignoring remaining %v", requiredDuration.Round(time.Second))
+			return false
+		}
+
 		// remember last active plan's end time
 		lp.setPlanActive(true)
 		lp.planSlotEnd = slotEnd
-	} else if lp.planActive && ((lp.clock.Now().Before(lp.planSlotEnd) && !lp.planSlotEnd.IsZero()) ||
-		(lp.clock.Now().After(lp.targetTime) && !lp.targetTime.IsZero())) {
-		// if slot still not completed or past target time continue charging
-		active = true
+	} else if lp.planActive {
+		// planner was active (any slot, not necessarily previous slot) and charge goal has not yet been met
+		switch {
+		case lp.clock.Now().After(lp.targetTime) && !lp.targetTime.IsZero():
+			// if the plan did not (entirely) work, we may still be charging beyond plan end- in that case, continue charging
+			// TODO check when schedule is implemented
+			lp.log.DEBUG.Println("continuing after target time")
+			active = true
+		case lp.clock.Now().Before(lp.planSlotEnd) && !lp.planSlotEnd.IsZero():
```

Comment:
/cc @schenlap ich kann mich nicht mehr erinnern, warum die das eingebaut hatten?

Reply:
wir wollen einen Slot in dem wir bereits gestartet haben nicht mehr abschalten.